File: java-junit-maven/src/test/java/varar/RomanNumeralsSteps.java
package varar;

import dev.varar.State;
import dev.varar.StepDefinitions;
import dev.varar.Steps;
import examples.RomanNumerals;
import java.util.Map;

public final class RomanNumeralsSteps implements StepDefinitions<RomanNumeralsSteps.Ctx> {

    record Ctx() implements State {}

    @Override
    public void register(Steps<Ctx> s) {
        s.state(Ctx::new);

        s.sensor(
                "a decimal and a roman number",
                (Ctx ctx, Map<String, String> row) -> Map.of(
                        "decimal", row.get("decimal"),
                        "roman", RomanNumerals.toRoman(Integer.parseInt(row.get("decimal")))));
    }
}

File: kotlin-junit/src/test/kotlin/varar/roman-numerals.steps.kt
@file:JvmName("RomanNumeralsSteps")

package varar

import dev.varar.kotlin.sensor
import dev.varar.kotlin.steps
import examples.toRoman

val romanNumeralsSteps = steps {
    sensor("a decimal and a roman number") { row: Map<String, String> ->
        mapOf(
            "decimal" to row.getValue("decimal"),
            "roman" to toRoman(row.getValue("decimal").toInt()),
        )
    }
}
